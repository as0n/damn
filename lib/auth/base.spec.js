jest.mock('../client', () => jest.fn())

const AuthenticationStrategy = require('./base')
const HttpClient = require('../client')

describe('AuthenticationStrategy', () => {
  const baseUrl = 'qdqmdsk'
  let baseAuth = null

  beforeEach(() => {
    HttpClient.mockClear()
    baseAuth = new AuthenticationStrategy(baseUrl)
  })

  describe('constructor(baseURL)', () => {
    it('instanciate a client with the base URL', () => {
      expect(baseAuth).toHaveProperty('client')

      expect(HttpClient).toHaveBeenCalledTimes(1)
      expect(HttpClient.mock.calls[0][1]).toBe(baseUrl)
    })

    it('has a default base URL', () => {
      baseAuth = new AuthenticationStrategy()
      expect(baseAuth).toHaveProperty('client')

      expect(HttpClient).toHaveBeenCalledTimes(2)
      expect(HttpClient.mock.calls[1][1]).toBe('')
    })
  })

  describe('getToken()', () => {
    it('should not be implemented', () => {
      expect(baseAuth.getToken).toThrowErrorMatchingSnapshot()
    })
  })
})
