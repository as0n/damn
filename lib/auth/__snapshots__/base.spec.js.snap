// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthenticationStrategy getToken() should not be implemented 1`] = `"Implement me!"`;
